tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AWS/SNS/ContextPropagation.cs

```diff
@@ -27,14 +28,34 @@ private static void Inject<TMessageRequest>(SpanContext context, IDictionary mes
             sb.Append('}');
 
             var resultString = Util.StringBuilderCache.GetStringAndRelease(sb);
-            byte[] bytes = Encoding.UTF8.GetBytes(resultString);
-            MemoryStream stream = new MemoryStream(bytes);
+            var bytes = Encoding.UTF8.GetBytes(resultString);
+            var stream = new MemoryStream(bytes);
             messageAttributes[SnsKey] = CachedMessageHeadersHelper<TMessageRequest>.CreateMessageAttributeValue(stream);
         }
 
-        public static void InjectHeadersIntoMessage<TMessageRequest>(IContainsMessageAttributes carrier, SpanContext spanContext)
+        public static void InjectHeadersIntoBatch<TBatchRequest>(IPublishBatchRequest request, SpanContext context)
```

Comment:
To avoid the boxing, you should use a `where` constraint 🙂 
```suggestion
        public static void InjectHeadersIntoBatch<TBatchRequest>(TBatchRequest request, SpanContext context)
            where TBatchRequest : IPublishBatchRequest
```

Reply:
Resolving due to [this comment](https://github.com/DataDog/dd-trace-dotnet/pull/4712#issuecomment-1762212524).